lib/rubocop/cop/rspec/capybara/specific_matcher.rb

```diff
@@ -28,6 +28,24 @@ class SpecificMatcher < Base
           MSG = 'Prefer `%<good_matcher>s` over `%<bad_matcher>s`.'
           RESTRICT_ON_SEND = %i[have_selector have_no_selector have_css
                                 have_no_css].freeze
+          COMMON_OPTIONS = %w[
+            above below left_of right_of near count minimum maximum between text
+            id class style visible obscured exact exact_text normalize_ws match
+            wait filter_set focused
+          ].freeze
+          HAVE_BUTTON_OPTIONS = COMMON_OPTIONS + %w[
+            disabled name value title type
+          ].freeze
```

Comment:
I just realized that this way, `HAVE_BUTTON_OPTIONS` is not frozen. We’d have to do e.g.
```rb
HAVE_BUTTON_OPTIONS = (COMMON_OPTIONS + %w[
  disabled name value title type
]).freeze
```
which really doesn’t seem very elegant.

Perhaps we should just have one big SPECIFIC_MATCHER_OPTIONS hash instead? What do you think @ydah?

```rb
SPECIFIC_MATCHER_OPTIONS = {
  'button' => COMMON_OPTIONS + %w[
    disabled name value title type
  ],
  'link' => COMMON_OPTIONS + %w[
    href alt title download
  ],
  'table' => COMMON_OPTIONS + %w[
    caption with_cols cols with_rows rows
  ],
  'select' => COMMON_OPTIONS + %w[
    disabled name placeholder options enabled_options disabled_options
    selected with_selected multiple with_options
  ]
}.freeze
```

Reply:
You'd still have to freeze the hash values, but that can be done like this:
```ruby
SPECIFIC_MATCHER_OPTIONS = {
  #...
}.each_value(&:freeze).freeze
```

Reply:
In that case, I think I’d prefer calling `freeze` directly on the created arrays anyway:
```rb
SPECIFIC_MATCHER_OPTIONS = {
  'button' => (
    COMMON_OPTIONS + %w[disabled name value title type]
  ).freeze,
  'link' => (
    COMMON_OPTIONS + %w[href alt title download]
  ).freeze,
  'table' => (
    COMMON_OPTIONS + %w[caption with_cols cols with_rows rows]
  ).freeze,
  'select' => (
    COMMON_OPTIONS + %w[
      disabled name placeholder options enabled_options
      disabled_options selected with_selected multiple with_options
    ]
  ).freeze
}.freeze
```

Reply:
My apologies for merging this. I've just realized there are more comments added after I last refreshed the page.
I hope it's not something that can't be fixed in a follow-up.